pkg/datapath/fake/node.go

```diff
@@ -55,6 +55,13 @@ func (n *FakeNodeHandler) AllocateNodeID(_ net.IP) uint16 {
 	return 0
```

Comment:
I guess its plausible that we could see the nodeDelete before pod delete events and continue to send traffic. Then without a nodeID mapping we would see policy block errors because the traffic would be dropped.

I really wanted to simply push all the handling of nodeIDs into ipcache logic so that it is not managed through addNode at all. WDYT? I hesitated thinking this promote logic would ensure we dropped it on node delete.  But, perhaps its better to just drop the node logic.